### PR TITLE
Malte lig 2469 pass datasetcreator when creating a

### PR DESCRIPTION
**Pull request by MalteEbner**
Pip part of the issue, the docker part will be based on this.

## Description
Added the DatasetCreator argument to
- ApiWorkflowClient.create_dataset
- ApiWorkflowClient.create_new_dataset_with_unique_name

And added unittests for the new argument.

## Other
- generated new openapi specs
- autoformatted the two changed files with black

### CODE REVIEW

File: lightly/api/api_workflow_datasets.py [hunk 1]
```diff
@@ -165,8 +161,12 @@ def set_dataset_id_by_name(self, dataset_name: str, shared: Optional[bool] = Fal
                 )
             warnings.warn(msg)
 
-
-    def create_dataset(self, dataset_name: str, dataset_type: Optional[str] = None):
+    def create_dataset(
+        self,
+        dataset_name: str,
+        dataset_type: str = DatasetType.IMAGES,
+        dataset_creator: Optional[DatasetCreator] = DatasetCreator.USER_PIP,
```
Comment (guarin) [inline lightly/api/api_workflow_datasets.py:168]: I would prefer to save the creator as an instance attribute on the class. Then we can easily change it from the worker. Otherwise we have to check all functions to make sure we didn't forget to set the creator. I already discussed this with @michal-lightly and we'll do the same for the different users (lig-2466).

**Review by guarin — COMMENTED** [review]
The code looks good to me. There is just one question on whether we should store the creator as an instance attribute.

File: lightly/api/api_workflow_datasets.py [hunk 2]
```diff
@@ -6,35 +6,33 @@
     CreateEntityResponse,
```
Comment (MalteEbner) [inline lightly/api/api_workflow_datasets.py:6]: This file was formatted with black.

File: lightly/api/api_workflow_datasets.py [hunk 3]
```diff
@@ -230,14 +231,16 @@ def _create_dataset_without_check_existing(
                 constants `DatasetType.IMAGES` and `DatasetType.VIDEOS`.
 
         """
-        body = DatasetCreateRequest(name=dataset_name, type=dataset_type)
+        body = DatasetCreateRequest(
+            name=dataset_name, type=dataset_type, creator=self.dataset_creator
```
Comment (MalteEbner) [inline lightly/api/api_workflow_datasets.py:235]: The relevant change

File: lightly/api/api_workflow_datasets.py [hunk 4]
```diff
@@ -230,14 +231,16 @@ def _create_dataset_without_check_existing(
...
+        )
         response: CreateEntityResponse = self._datasets_api.create_dataset(body=body)
         self._dataset_id = response.id
 
     def create_new_dataset_with_unique_name(
         self,
         dataset_basename: str,
-        dataset_type: Optional[str] = None,
+        dataset_type: str = DatasetType.IMAGES,
```
Comment (MalteEbner) [inline lightly/api/api_workflow_datasets.py:243]: An unrelated change

File: tests/api_workflow/test_api_workflow_datasets.py [hunk 5]
```diff
@@ -1,10 +1,18 @@
+from pytest_mock import MockerFixture
```
Comment (MalteEbner) [inline tests/api_workflow/test_api_workflow_datasets.py:1]: This file was formatted with black as well.

File: tests/api_workflow/test_api_workflow_datasets.py [hunk 6]
```diff
@@ -124,6 +171,19 @@ def test_dataset_type(self):
...
+    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
+    client = ApiWorkflowClient()
+    client.dataset_creator = DatasetCreator.USER_PIP
+    client._datasets_api = mocker.create_autospec(DatasetsApi)
+
+    client.create_dataset(dataset_name="name")
+    expected_body = DatasetCreateRequest(
+        name="name", type=DatasetType.IMAGES, creator=DatasetCreator.USER_PIP
+    )
+    client._datasets_api.create_dataset.assert_called_once_with(expected_body)
```
Comment (MalteEbner) [inline tests/api_workflow/test_api_workflow_datasets.py:189]: The new test

File: lightly/api/api_workflow_client.py [hunk 7]
```diff
@@ -75,13 +75,16 @@ class ApiWorkflowClient(_UploadEmbeddingsMixin,
...
+            Telling from where the dataset is created.
     """
 
     def __init__(
         self,
         token: Optional[str] = None,
         dataset_id: Optional[str] = None,
-        embedding_id: Optional[str] = None
+        embedding_id: Optional[str] = None,
+        dataset_creator: DatasetCreator = DatasetCreator.USER_PIP,
```
Comment (guarin) [inline lightly/api/api_workflow_client.py:87]:
```suggestion
        dataset_creator: str = DatasetCreator.USER_PIP,
```

File: lightly/api/api_workflow_client.py [hunk 8]
```diff
@@ -105,6 +108,7 @@ def __init__(
             self._dataset_id = dataset_id
         if embedding_id is not None:
             self.embedding_id = embedding_id
+        self.dataset_creator = dataset_creator
```
Comment (guarin) [inline lightly/api/api_workflow_client.py:111]: Did you consider making the attribute private? In my eyes it should not be changed after the client is created.

Comment (MalteEbner) [reply lightly/api/api_workflow_client.py]: Good point, I changed it.

**Review by guarin — APPROVED** [review]
Thanks! LGTM!
